Argo/Functions/decode.swift

```diff
@@ -1,35 +1,215 @@
-// MARK: decode root objects and arrays
+/**
+  Decode a raw JSON value into an optional `Decodable` value
 
+  This function takes the output from `NSJSONSerialization` and attempts to
+  decode it into a `Decodable` value, returning an `Optional`. This works based
+  on the type you ask for.
+
+  For example, the following code attempts to decode an optional `String`,
+  because that's what we have explicitly stated is the return type:
+
+  ```
+  do {
+    let object = try NSJSONSerialization.JSONObjectWithData(data, options: nil)
+    let str: String? = decode(object)
+  } catch {
+    // handle error
+  }
+  ```
+
+  - parameter object: The output from `NSJSONSerialization`
+  - returns: An `Optional` value of type `T` where `T` is `Decodable`
+*/
 public func decode<T: Decodable where T == T.DecodedType>(object: AnyObject) -> T? {
   return decode(object).value
 }
 
+/**
+  Decode a raw JSON value into an optional array of `Decodable` values
```

Comment:
How do you feel about using "Parse" instead of Decode at the beginning of this sentence (or really any sentence that mentions the `Decodable` type)? Maybe it's just me but I have to double take on every sentence that uses Decode twice like this.